src/slices/sort.go

```diff
@@ -8,14 +8,16 @@ package slices
 
 import (
 	"cmp"
+	"internal/reflectlite"
 	"math/bits"
 )
 
 // Sort sorts a slice of any ordered type in ascending order.
 // When sorting floating-point numbers, NaNs are ordered before other values.
 func Sort[S ~[]E, E cmp.Ordered](x S) {
-	n := len(x)
-	pdqsortOrdered(x, 0, n, bits.Len(uint(n)))
+	sortSlice(x, func(i, j int) bool {
+		return x[i] < x[j]
```

Comment:
Shouldn't that be:

```suggestion
		return cmp.Compare(x[i], x[j]) < 0
```
?

Reply:
Yeah, I suppose if one cares about floating point NaN comparison ;) 

And sure enough, the tests care.